### ISSUE TITLE
Have torch.manual_seed seed all GPUs as well

### ISSUE THREAD

**Issue by apaszke** [pytorch/pytorch#1615]
It's a bit tricky, because it will need to seed them lazily

**Comment by colesbury** [comment]
It's already lazy:

```
import torch
torch.cuda.manual_seed_all(1)
```

https://github.com/torch/cutorch/commit/8b349851f431d49015040dfc0ed4938c583e7ce1

**Comment by alykhantejani** [comment]
@apaszke I can send a PR for this, I took a quick look and seems like all that is needed is to add the following to `torch.manual_seed()`
```python
if torch.cuda.is_available:
    torch.cuda.manual_seed_all(seed)
```

However I noticed the `torch.cuda.manual_seed*` functions don't return a generator, whereas `torch.manual_seed` does. Is there a reason for this?

**Comment by apaszke** [comment]
Yes I think it's enough to add these two lines.

It's because there are no CUDA generators. There's only a single cuRAND state per device

**Comment by cy-xu** [comment]
Hi, a follow up on a different perspective. I manual seed both pytorch and numpy, and tested on a single GPU to get matching random numbers. But I train on the cluster and the scheduler assigns me a to a random GPU each time, then I found that numpy still produce identical numbers across different GPU but numpy's random number is different on different cards... Did anyone notice this before?
